packages/xstate-solid/src/createImmutable.ts

```diff
@@ -0,0 +1,114 @@
+import { createStore, Store } from 'solid-js/store';
+import { deepClone } from './deepClone';
+import { batch } from 'solid-js';
+
+export function isWrappable(obj: any): obj is object {
+  let proto;
+  return (
+    obj != null &&
+    typeof obj === 'object' &&
+    (!(proto = Object.getPrototypeOf(obj)) ||
+      proto === Object.prototype ||
+      Array.isArray(obj))
+  );
+}
+
+const resolvePath = (path: any[], obj = {}) =>
+  path.reduce((prev, curr) => prev?.[curr], obj);
```

Comment:
for perf reasons I would probably rewrite this using a simple for loop:
```suggestion
const resolvePath = (path: any[], obj = {}) => {
  let current = obj;
  for (let i = 0; i < path.length; i++) {
    current = current?.[path[i]];
  }
  return current;
}
```